### PR TITLE
chore: Add C examples

### PR DESCRIPTION
**Pull request by jubianchi**
* [x] Add basic examples
* [x] Add command to test the examples
* [x] Update docs.wasmer.io with the new API changes. Namely: https://github.com/wasmerio/wasmer/pull/1871/commits/4df7722f253b249e6ed26fe7f89a8c69c6bc60d1

# Review

- [ ] Add a short description of the the change to the CHANGELOG.md file

### CODE REVIEW

**Comment by jubianchi** [comment]
bors try

**Comment by jubianchi** [comment]
bors try-

**Comment by jubianchi** [comment]
bors try

**Review by syrusakbary — APPROVED** [review]
Good to merge once tests pass!

File: Makefile [hunk 1]
```diff
@@ -246,7 +246,7 @@ test-capi-llvm-native:
 	cargo test --manifest-path lib/c-api/Cargo.toml --release \
 		--no-default-features --features deprecated,wat,native,llvm,wasi $(capi_default_features) -- --nocapture
 
-test-capi-examples:
+test-capi-examples: package-capi
```
Comment (syrusakbary) [inline Makefile:249]: Oops!

Comment (jubianchi) [reply Makefile:249]: @syrusakbary i did that on purpose: the examples expect to find the `include/` and `lib/` directory in `WASMER_DIR`.

After the library has been built by the regular tests, it's in `target/release` without the expected structure.

I know this is not ideal: it makes the `test-capi-examples` target implicitely dependant to `build-capi`

**Comment by syrusakbary** [comment]
bors try

**Comment by syrusakbary** [comment]
bors try-

File: Makefile (hunk 1 above)
Comment (syrusakbary) [reply Makefile:251]: Oh! The Oops was meant for myself for forgetting that :P

**Comment by Hywan** [comment]
:+1:

**Pull request merged**